src/ast/implementation/declaration/user_defined_value_type_definition.ts

```diff
@@ -0,0 +1,58 @@
+import { ASTNode } from "../../ast_node";
+import { SourceUnit } from "../meta/source_unit";
+import { ElementaryTypeName } from "../type/elementary_type_name";
+import { ContractDefinition } from "./contract_definition";
+
+export class UserDefinedValueTypeDefinition extends ASTNode {
+    /**
+     * The name of the user-defined value type definition
+     */
+    name: string;
+
+    /**
+     * Canonical name (or qualified name), e.g. `DefiningContract.SomeType`.
+     * Is `undefined` for Solidity 0.8.8. Available since Solidity 0.8.9.
+     */
+    canonicalName?: string;
```

Comment:
Maybe in a compute this property for 0.8.8 so we can always assume its present?

Reply:
Does not seem too complex to fix in-place. Will look into.

Reply:
The thing here is that this node is constructed before "access to parent" is available. So it would require a postprocessing pass to fix this. I'm arguing if it would be easier to just drop `canonicalName` and make it a getter instead due to field value should change if parent changes. Currently it is just static value from compiler itself and does not react on tree modification. However, this would involve much greater cleanup, as `canonicalName` is available in few other type nodes (structs, enums and whatever else). So lets keep current state of things for now, then do a cleanup pass to compyte such things on runtime.

Reply:
Created issue #80 to track this.